lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -206,6 +208,24 @@ export class AuthCache extends BrowserStorage {// Singleton
         return results;
     }
 
+    /**
+     * Get idToken from the cache
+     */
+    getIdToken(clientId: string, homeAccountIdentifier: string, authority: string): AccessTokenCacheItem {
+        const idTokenKey: AccessTokenKey = {
+            authority: authority,
+            clientId: clientId,
+            scopes: undefined,
+            homeAccountIdentifier: homeAccountIdentifier
+        };
+
+        const idTokenKeyString = JSON.stringify(idTokenKey);
+        const idToken = this.getItem(idTokenKeyString);
+        const idTokenValue = JSON.parse(idToken) as AccessTokenValue;
```

Comment:
What should happen if `JSON.parse` throws?

Reply:
IMO this should be in a try/catch and have a logger message + return null if it throws

Reply:
Implemented try/catch + returning null. The logger message was already there after the call to `getIdToken` in `acquireTokenSilent`. Updated the logic around the call as well to handle this possible exception.